### ISSUE TITLE
BUG: xgboost pipeline failing due to deprecated use_label_encoder

### ISSUE THREAD

**Issue by romanlutz** [fairlearn/fairlearn#1203]
#### Describe the bug

Currently, the "other ML" package tests are failing, e.g.: https://github.com/fairlearn/fairlearn/actions/runs/4094942670/jobs/7061519839
The offending line is:
```
estimator = xgb.XGBClassifier(use_label_encoder=False)
```
since `use_label_encoder` is deprecated in 1.7.0.

For context, latest xgboost is 1.7.3 as of February 4, 2023. Apparently, this is now the default behavior since sklearn deprecated the label encoder.

Maybe this can be fixed by checking for the xgboost version and if it's `>=1.7` then don't provide the argument (?)

#### Steps/Code to Reproduce

```
estimator = xgb.XGBClassifier(use_label_encoder=False)
```

#### Expected Results

No warnings!

#### Actual Results
```
def test_expgrad_classification():
>       estimator = xgb.XGBClassifier(use_label_encoder=False)

test_othermlpackages/test_xgboost.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/share/miniconda/envs/test-conda-env/lib/python3.8/site-packages/xgboost/core.py:620: in inner_f
    return func(**kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'XGBClassifier' object has no attribute 'objective'") raised in repr()] XGBClassifier object at 0x7f3e159698e0>
objective = 'binary:logistic', use_label_encoder = False, kwargs = {}

    @_deprecate_positional_args
    def __init__(
        self,
        *,
        objective: _SklObjective = "binary:logistic",
        use_label_encoder: Optional[bool] = None,
        **kwargs: Any,
    ) -> None:
        # must match the parameters for `get_params`
        self.use_label_encoder = use_label_encoder
        if use_label_encoder is True:
            raise ValueError("Label encoder was removed in 1.6.0.")
        if use_label_encoder is not None:
>           warnings.warn("`use_label_encoder` is deprecated in 1.7.0.")
E           UserWarning: `use_label_encoder` is deprecated in 1.7.0.

/usr/share/miniconda/envs/test-conda-env/lib/python3.8/site-packages/xgboost/sklearn.py:1395: UserWarning
________________________ test_gridsearch_classification ________________________

    def test_gridsearch_classification():
>       estimator = xgb.XGBClassifier(use_label_encoder=False)

test_othermlpackages/test_xgboost.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/share/miniconda/envs/test-conda-env/lib/python3.8/site-packages/xgboost/core.py:620: in inner_f
    return func(**kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'XGBClassifier' object has no attribute 'objective'") raised in repr()] XGBClassifier object at 0x7f3dfe3254f0>
objective = 'binary:logistic', use_label_encoder = False, kwargs = {}

    @_deprecate_positional_args
    def __init__(
        self,
        *,
        objective: _SklObjective = "binary:logistic",
        use_label_encoder: Optional[bool] = None,
        **kwargs: Any,
    ) -> None:
        # must match the parameters for `get_params`
        self.use_label_encoder = use_label_encoder
        if use_label_encoder is True:
            raise ValueError("Label encoder was removed in 1.6.0.")
        if use_label_encoder is not None:
>           warnings.warn("`use_label_encoder` is deprecated in 1.7.0.")
E           UserWarning: `use_label_encoder` is deprecated in 1.7.0.

/usr/share/miniconda/envs/test-conda-env/lib/python3.8/site-packages/xgboost/sklearn.py:1395: UserWarning
____________________ test_thresholdoptimizer_classification ____________________

    def test_thresholdoptimizer_classification():
>       estimator = xgb.XGBClassifier(use_label_encoder=False)

test_othermlpackages/test_xgboost.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/share/miniconda/envs/test-conda-env/lib/python3.8/site-packages/xgboost/core.py:620: in inner_f
    return func(**kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'XGBClassifier' object has no attribute 'objective'") raised in repr()] XGBClassifier object at 0x7f3dfe6b[89](https://github.com/fairlearn/fairlearn/actions/runs/4094942670/jobs/7061519839#step:7:90)70>
objective = 'binary:logistic', use_label_encoder = False, kwargs = {}

    @_deprecate_positional_args
    def __init__(
        self,
        *,
        objective: _SklObjective = "binary:logistic",
        use_label_encoder: Optional[bool] = None,
        **kwargs: Any,
    ) -> None:
        # must match the parameters for `get_params`
        self.use_label_encoder = use_label_encoder
        if use_label_encoder is True:
            raise ValueError("Label encoder was removed in 1.6.0.")
        if use_label_encoder is not None:
>           warnings.warn("`use_label_encoder` is deprecated in 1.7.0.")
E           UserWarning: `use_label_encoder` is deprecated in 1.7.0.

/usr/share/miniconda/envs/test-conda-env/lib/python3.8/site-packages/xgboost/sklearn.py:13[95](https://github.com/fairlearn/fairlearn/actions/runs/4094942670/jobs/7061519839#step:7:96): UserWarning
```
#### Versions

see link

**Comment by romanlutz** [comment]
Closing since I fixed this with #1204
